src/coreComponents/mesh/MeshManager.cpp

```diff
@@ -43,15 +43,15 @@ MeshManager::~MeshManager()
 Group * MeshManager::createChild( string const & childKey, string const & childName )
 {
   GEOS_LOG_RANK_0( "Adding Mesh: " << childKey << ", " << childName );
-  std::unique_ptr< MeshGeneratorBase > solver = MeshGeneratorBase::CatalogInterface::factory( childKey, childName, this );
-  return &this->registerGroup< MeshGeneratorBase >( childName, std::move( solver ) );
+  std::unique_ptr< MeshBase > mesh = MeshBase::CatalogInterface::factory( childKey, childName, this );
```

Comment:
Now `MeshManager` depends on `MeshBase` and `MeshGeneratorBase`.
Couldn't we have one single concept and use it? I'd say that would make the things clearer.

Reply:
Is there an answer to this question? To me, if `MeshBase` and `MeshGeneratorBase` have different responsibilities and `MeshGeneratorBase` is intended to generate something derived from `MeshBase`, then I don't see a problem....but perhaps I don't understand.